src/u_scanner.c

```diff
@@ -216,6 +216,48 @@ static void U_RestoreState(u_scanner_t* s, u_scanner_t savedstate)
   }
 }
 
+boolean U_GetString(u_scanner_t* scanner)
+{
+  unsigned int start;
+  unsigned int end;
+  char cur;
+  u_parserstate_t* nextState = &scanner->nextState;
+
+  if(!scanner->needNext)
+  {
+    scanner->needNext = true;
+    U_ExpandState(scanner);
+    return true;
+  }
+
+  nextState->tokenLine = scanner->line;
+  nextState->tokenLinePosition = scanner->scanPos - scanner->lineStart;
+  nextState->token = TK_NoToken;
+  if(scanner->scanPos >= scanner->length)
+  {
+    U_ExpandState(scanner);
+    return false;
+  }
+
+  start = scanner->scanPos;
+  cur   = scanner->data[scanner->scanPos++];
+
+  while(scanner->scanPos < scanner->length)
+  {
+    cur = scanner->data[scanner->scanPos];
+
+    if(cur != ' ' && cur != '\t' && cur != 0 && (cur != '\n' || cur != '\r'))
```

Comment:
I don't understand that last condition. We want to call `break` if `cur` is either `'\n'` or `'\r'` (or `' '` or `'\t'` or `'\0'`), right? Maybe call `isgraph()` here for unambiguity? 😉

Reply:
Well in the same way we check for whitespace in `U_CheckForWhitespace`.

Reply:
This code is really confusing at times...

Reply:
Yes, I need to test it more. Maybe we should refactor the whitespace check.

Reply:
Well, I consider any string that contains control characters other than the "usual few" (`\t`, `\r`, `\n`) as malformed and demand any right to fail in one way or the other. 😁